pkg/netdevice/netDeviceProvider.go

```diff
@@ -104,16 +106,16 @@ func hasDefaultRoute(pciAddr string) (bool, error) {
 	}
 
 	if len(ifNames) > 0 { // there's at least one interface name found
-		for _, ifName := range ifNames {
-			link, err := netlink.LinkByName(ifName)
+		for iIfName := range ifNames {
```

Comment:
not really a fan of this, do you recall which CI check complains for the string copy here ?
i dont think these memory optimizations should be a strict requirement.

Reply:
Linter message:
```hugeParam: X is heavy (C bytes); consider passing it by pointer (gocritic)``` 

yeah - I did it just to silence the CI check output. Definitely agree it shouldn't be a strict requirement since this software is just a control plane component and it's dealing with objects in memory that are relatively small mem footprint (~300bytes) and are few in number.

Do you not like it because it reduces readability? I think it's fine but happy to revert if you want.

Reply:
> Do you not like it because it reduces readability?

yes exactly. so essentially i prefer to ignore this check or make it less restrictive so it would work in most cases.

LMK if you have other thoughts on this, its just my opinion here :) and ill be fine to keep it as is if you think its important to have this check enforced.


perhaps just set a bigger threshold for this.

from gocritic docs:
> `@hugeParam.sizeThreshold` size in bytes that makes the warning trigger (default 80)

so maybe set this to a higher number ? (512?)

Reply:
Done